File: frontend/src/types/batch.ts
export interface Batch {
  id: string;
  name: string;
  description: string | null;
  created_by: string;
  total_documents: number;
  processed_documents: number;
  failed_documents: number;
  created_at: string;
  updated_at: string;
}

export interface BatchCreate {
  name: string;
  description?: string;
}

File: backend/app/schemas/batch.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class BatchCreate(BaseModel):
    name: str
    description: str | None = None


class BatchOut(BaseModel):
    id: uuid.UUID
    name: str
    description: str | None
    created_by: uuid.UUID
    total_documents: int
    processed_documents: int
    failed_documents: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
